tiledb/sm/query/readers/sparse_index_reader_base.h

```diff
@@ -514,16 +543,16 @@ class SparseIndexReaderBase : public ReaderBase {
   /**
    * Returns the current read state.
    *
-   * @return pointer to the read state.
+   * @return reference to the read state.
    */
-  const ReadState* read_state() const;
+  const ReadState& read_state() const;
 
   /**
    * Returns the current read state.
    *
-   * @return pointer to the read state.
+   * @return reference to the read state.
    */
-  ReadState* read_state();
+  ReadState& read_state();
```

Comment:
Audit the code and see if the only reason we need this is because we want to update the read state. If so, let's rename to void set_read_state(...)